Switch to MDA rtd sphinx theme

ojeda-e opened the pull request:
As we agreed with @IAlibay, I am here submitting changes to update the docs theme to the official one. This PR fixes #120 and #110 (opened by @orbeckst).

## Description
Update docs `mdanalysis-sphinx-theme` with plus maintenance.

## Status
- [x] Ready to go

It compiles locally and seems to work as expected. I'll be fixing the issues if any once it build in the CI.
Thanks.

ojeda-e committed:
Update to mdanalysis-sphinx-theme

ojeda-e committed:
Docs: Fix broken link

ojeda-e committed:
Build: Update docs dependencies.

ojeda-e committed:
Docs: Fix math syntax in new theme.

ojeda-e committed:
Docs: Rerun and checked tutorials. Working.

ojeda-e committed:
Docs: Finalize switch to mda rtd theme

ojeda-e committed:
Build: Add missing dep

ojeda-e committed:
Docs: rerun nb with default kernel.

ojeda-e committed:
Update CHANGELOG

orbeckst reviewed (commented):
Thanks for link fixing and face-lifting.

Is the content from the removed notebooks available elsewhere?

ojeda-e commented:
> Is the content from the removed notebooks available elsewhere?

@orbeckst I just noticed the diff for the notebooks are misleading. The notebooks are still part of the documentation. I had to regenerate them to verify every cell works and the output is correct -the same as before-; the nb kernel was also updated to enable docs build in the CI.
The content is available in the reviewnb app: https://app.reviewnb.com/MDAnalysis/membrane-curvature/pull/122/

Here are the build notebooks in the tutorial section in case you want to check:
- https://membrane-curvature--122.org.readthedocs.build/en/122/source/pages/Curvature_membrane-only_systems.html
- https://membrane-curvature--122.org.readthedocs.build/en/122/source/pages/Curvature_membrane_protein_nopr_all_atom.html

orbeckst reviewed (approved):
LGTM, thanks for putting in the effort!

ojeda-e commented:
@lilyminium, if possible, would you please help me with a quick review here before I merge?

ojeda-e committed:
Docs: Tidy output cells in nb.

ojeda-e committed:
Docs: Tidy output cells in nb; attempt

IAlibay commented:
@ojeda-e apologies I am unlikely to have much time to look at this until the weekeend - please do re-ping me if you still need my input by then.

ojeda-e committed:
Docs: Revert nb to original.

lilyminium commented on docs/README.md:
```diff
@@ -1,24 +1,46 @@
+To compile the docs, first ensure that:
 
+1. All necessary dependencies are installed.
+1. The editable version of this repo is installed.
```
This isn't necessarily a requirement -- so long as the docs don't execute code that's changed, I think you could get by with a non-editable install :-)

lilyminium commented on docs/README.md:
```diff
@@ -1,24 +1,46 @@
-be inside a directory called `html/` depending on what version of Sphinx is installed).
+## Installing the editable version
 
+Please souble check you have installed the editabe version of the package. **From the root of the project**, run:
```
```suggestion
Please double check you have installed the editable version of the package. **From the root of the project**, run:
```

Although again as noted above, this may not be essential.

lilyminium commented on docs/README.md:
````diff
@@ -1,24 +1,46 @@
+make html
+```
+
+The compiled docs will be in the `_build` directory and can be viewed by opening `index.html` (which may itself be inside a directory called `html/` depending on what version of Sphinx is installed).
````
Ah, does this depend on the Sphinx version? An alternative is to explicitly call [sphinx-build](https://www.sphinx-doc.org/en/master/man/sphinx-build.html) which requires input and output directories to be specified, instead of `make html`.

ojeda-e replied:
Thanks for catching this, it comes from a very old version >.< Deleted.

lilyminium commented on docs/README.md:
```diff
@@ -1,24 +1,46 @@
+
+The compiled docs will be in the `_build` directory and can be viewed by opening `index.html` (which may itself be inside a directory called `html/` depending on what version of Sphinx is installed).
+
+A configuration file for [Read The Docs](https://readthedocs.org/) (readthedocs.yaml) is included in the top level of the repository. To use Read the Docs to host your documentation, go to https://readthedocs.org/ and connect this repository. You may need to change your default branch to `main` under Advanced Settings for the project.
```
I'm not sure this documentation is totally necessary either -- it seems like it would be quite rare for a user to want to host documentation of Membranecurvature?

ojeda-e replied:
Yeah, this came from the cookie cutter template. Will remove this. Thanks.

lilyminium commented on docs/requirements.txt:
```diff
@@ -0,0 +1,14 @@
+MDAnalysisData
+networkx # LeafletFinder dependency
+tidynamics # MSD dependency
+sphinx <7.0
```
I know we pinned this for a bit, but I think we've finally been able to unpin Sphinx and work with the most recent versions. Is `sphinx<7.0` still needed here?

ojeda-e replied:
Thanks, actually now I realized I can fully skipped `Sphinx` here because `sphinx>=6.2.1` is coming as transient dep from `mdanalysis-sphinx-theme` anyway. :)  I'll delete line 9 here.

lilyminium reviewed (commented):
Looks pretty good @ojeda-e, I've just left some comments largely on the notes. I also think you've put last year in the "Last executed" line of ` docs/source/pages/Curvature_membrane-only_systems.ipynb` (ReviewNB won't let me comment directly....)

Just commenting this review to avoid blocking progress :-)

ojeda-e committed:
Other: Address comments from reviewer in README and requirements

ojeda-e committed:
Docs: Fix date on notebook

ojeda-e commented:
Thanks for taking the time here @lilyminium
I addressed your comments. Hopefully I didn't miss anything. Removed the editable version part, removed Sphinx from the requirements and fixed the date on the notebook.It's a bit late here, but there if there are more comments please feel free to leave them and I'll address them tomorrow.
